src/parser.ts

```diff
@@ -0,0 +1,468 @@
+  }
+  const [proofNquads, docNquads] = parts;
+  const docLines = docNquads.split(/\n/);
+  const lastLine = docLines.pop();
+  if (lastLine !== "") {
+    throw new Error(
+      `Expected empty last line, but found: ${JSON.stringify(lastLine)}`
+    );
+  }
+  const proofLines = proofNquads.split(/\n/);
+  const proofStatements = proofLines.map(parseQuad);
+  const docStatements = docLines.map(parseQuad);
+  const rdf = renderRdfLdpTable(proofStatements, docStatements);
+  const nquads = { proofNquads, docNquads };
+  const content = renderRdfLdpContent(proofStatements, docStatements);
+  return {
+    rdf,
+    nquads,
+    content,
+  };
+}
+
+function renderRdfLdpTable(proofStatements, docStatements) {
+  return {
+    document: renderRdfDatasetTable(docStatements),
+    proofOptions: renderRdfDatasetTable(proofStatements),
+  };
+}
+
+function renderRdfDatasetTable(statements): {
+  headers: Array<string>;
+  body: Array<Array<string | Pair>>;
+} {
+  let hasGraph = false;
+  for (const statement of statements) {
+    if (statement.graph) {
+      hasGraph = true;
+      break;
+    }
+  }
+
+  const headers: Array<string> = [];
+  const body: Array<Array<string | Pair>> = [];
+  headers.push("Subject");
+  headers.push("Predicate");
+  headers.push("Object");
+
+  if (hasGraph) {
+    headers.push("Graph");
+  }
+
+  for (const statement of statements) {
+    body.push(renderStatement(statement));
+  }
+
+  return {
+    headers,
+    body,
+  };
+}
+
+function renderDateFull(object): string {
+  if (object.type !== "literal") {
+    throw new Error(`Expected literal, got: ${object.type}`);
+  }
+  if (object.datatypeIri !== "http://www.w3.org/2001/XMLSchema#dateTime") {
+    throw new Error(
+      `Expected http://www.w3.org/2001/XMLSchema#dateTime, got ${object.datatypeIri}`
+    );
+  }
+
+  return new Date(object.value).toUTCString();
+}
+
+function renderCreated(object): { created: string } {
+  return {
+    created: renderDateFull(object),
+  };
+}
+
+interface Unknown {
+  unknown: string;
+}
+
+function renderProofPurpose(object): { purpose: string } | Unknown {
+  switch (object.value) {
+    case "https://w3id.org/security#assertionMethod":
+      return { purpose: "Assertion" };
+    default:
+      return { unknown: object.value };
+  }
+}
+
+function renderId(object): { id: string | Pair } {
+  return { id: renderTerm(object) };
+}
+
+function renderType(object): { type: string } {
+  if (object.type !== "iri") {
+    throw new Error("Expected IRI");
+  }
+  switch (object.value) {
+    case "https://w3id.org/security#TezosSignature2021":
+      return { type: "Tezos Signature 2021" };
+    case "https://www.w3.org/2018/credentials#VerifiableCredential":
+      return { type: "Verifiable Credential" };
+    case "https://tzprofiles.com/BasicProfile":
+      return { type: "Tezos Profile (tzprofiles.com)" };
+    default:
+      return { type: `Unkown type: ${object.value}` };
```

Comment:
Could this be rendered as a warning or error condition? Similar reason as for "Unrecognized terms" (https://github.com/spruceid/tzvm2021-explainer/pull/2/files#r669031310)

Reply:
Fixed in fb43edc09d3dcc6923f6b503959fb277984c4cdb, thanks